packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -608,33 +679,32 @@ function compileProofEntryConstraints(
           "Can't use isOwnerID and equalsEntry on the same entry."
         );
       }
-      entryEqualToOtherEntryByIndex.push(BigInt(firstOwnerIndex));
+      entryEqualToOtherEntryByIndex[entryIndex] = BigInt(firstOwnerIndex);
     } else if (entryInfo.entryConfig.equalsEntry !== undefined) {
       const otherEntryInfo = entryMap.get(entryInfo.entryConfig.equalsEntry);
       if (otherEntryInfo === undefined) {
         throw new Error(
           `Missing entry ${entryInfo.entryConfig.equalsEntry} for equality comparison.`
         );
       }
-      entryEqualToOtherEntryByIndex.push(BigInt(otherEntryInfo.entryIndex));
+      entryEqualToOtherEntryByIndex[entryIndex] = BigInt(
+        otherEntryInfo.entryIndex
+      );
     } else {
-      entryEqualToOtherEntryByIndex.push(BigInt(entryInfo.entryIndex));
+      entryEqualToOtherEntryByIndex[entryIndex] = BigInt(entryInfo.entryIndex);
     }
   }
 
-  // Equality constraints don't have an explicit disabled state, so spare
-  // entry slots always compare to themselves, to be a nop.
-  for (
-    let entryIndex = entryMap.size;
-    entryIndex < maxEntries + maxVirtualEntries;
-    entryIndex++
-  ) {
-    entryEqualToOtherEntryByIndex.push(BigInt(entryIndex));
-  }
-
   return {
     circuitEntryConstraintInputs: {
-      entryEqualToOtherEntryByIndex
+      // Equality constraints don't have an explicit disabled state, so spare
```

Comment:
I think this works, but I'm a bit nervous about the fact that it will now fill in undefined entries in the middle of the array, not only at the end.  I feel like that could hide issues with the loop above, since the fact that it will fill in all of the indexes is less obvious than it was before.  I feel like the old code made it obvious that it was filling in from the bottom (via push) and that what the loop at the end was doing was only padding out for extra circuit size.  This map is intended to do that, but could accidentally do more.